xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/js/xwiki/actionbuttons/actionButtons.js

```diff
@@ -23,359 +23,518 @@
 
 var XWiki = (function(XWiki) {
 // Start XWiki augmentation.
-var actionButtons = XWiki.actionButtons = XWiki.actionButtons || {};
-
-actionButtons.EditActions = Class.create({
-  initialize : function() {
-    this.addListeners();
-    this.addShortcuts();
-    this.addValidators();
-  },
-  addListeners : function() {
-    $$('input[name=action_cancel]').each(function(item) {
-      item.observe('click', this.onCancel.bindAsEventListener(this));
-    }.bind(this));
-    $$('input[name=action_preview]').each(function(item) {
-      item.observe('click', this.onSubmit.bindAsEventListener(this, 'preview'));
-    }.bind(this));
-    $$('input[name=action_save]').each(function(item) {
-      item.observe('click', this.onSubmit.bindAsEventListener(this, 'save'));
-    }.bind(this));
-    $$('input[name=action_saveandcontinue]').each(function(item) {
-      item.observe('click', this.onSubmit.bindAsEventListener(this, 'save', true));
-    }.bind(this));
-  },
-  addShortcuts : function() {
-    var shortcuts = {
-      'action_cancel' : "$services.localization.render('core.shortcuts.edit.cancel')",
-      'action_preview' : "$services.localization.render('core.shortcuts.edit.preview')",
-      // The following 2 are both "Back to edit" in the preview mode, depending on the used editor
-      'action_edit' : "$services.localization.render('core.shortcuts.edit.backtoedit')",
-      'action_inline' : "$services.localization.render('core.shortcuts.edit.backtoedit')",
-      'action_save' : "$services.localization.render('core.shortcuts.edit.saveandview')",
-      'action_propupdate' : "$services.localization.render('core.shortcuts.edit.saveandview')",
-      'action_saveandcontinue' : "$services.localization.render('core.shortcuts.edit.saveandcontinue')"
+  var actionButtons = XWiki.actionButtons = XWiki.actionButtons || {};
+
+  // we need to handle the creation of document
+  var currentDocument, currentVersion, isNew;
+  var editingVersionDate = new Date();
+
+  require(['xwiki-meta'], function (xm) {
+    currentDocument = xm.documentReference;
```

Comment:
I wonder if you should not wrap the whole file into this require() call. On the other hand, I doubt nobody will try to save a page before xwiki-meta is loaded.